rasa/utils/tensorflow/models.py

```diff
@@ -577,7 +577,7 @@ def _prepare_ffnn_layer(
     def _prepare_dot_product_loss(
```

Comment:
I'd argue that keeping a method with such a general name is a bit unfair now that it creates a single-label loss specifically. I know that we use single-label everywhere, but then one could ask why a model like IntentTED, which is so similar to TED, can't use a general method like this one, which TED uses... What do you think? Does it make sense to rename it to `_prepare_single_label_dot_product_loss` or to add an arg that determines which loss type will be used, so that this method can be used with both types?

Reply:
That will be handled in [Daksh's PR](https://github.com/RasaHQ/rasa/pull/8555/files#diff-1b0d53e90a7f1dcca1113cda22c58c98685004f8dfc7ba3d4e5bf9cee78e6a19R593)